packages/eas-build-job/src/job.ts

```diff
@@ -13,24 +13,38 @@ interface ImageMatchArgs {
   workflow: Workflow;
 }
 
-export const JobSchema = Joi.object<Job>({
+export const JobPlatfomSchema = Joi.object<Job>({
   platform: Joi.string()
     .valid(...Object.values(Platform))
     .required(),
-})
-  .when(Joi.object({ platform: Platform.ANDROID }).unknown(), { then: Android.JobSchema })
-  .when(Joi.object({ platform: Platform.IOS }).unknown(), { then: Ios.JobSchema });
+});
 
-export function sanitizeJob(
-  rawJob: object,
-  { sdkVersion, reactNativeVersion }: { reactNativeVersion?: string; sdkVersion?: string } = {}
-): Job {
-  const { value, error } = JobSchema.validate(rawJob, {
+function validateJobPlatform(rawJob: object): Platform {
```

Comment:
```suggestion
function sanitizeJobPlatform(rawJob: object): Platform {
```

Reply:
Usually: 
- `validate...` only validates the input, can throw an error, but does not return anything
- `sanitize...` validates the input, can throw an error, and returns the validated (and potentially changed) input